core/src/main/scala/io/chrisdavenport/shellfish/syntax/path/package.scala

```diff
@@ -0,0 +1,374 @@
+     */
+    def appendAs[A](contents: A)(implicit codec: Codec[A]): IO[Long] =
+      FilesOs.appendAs[A](path)(contents)
+
+  }
+
+  private val files = Files[IO]
+
+  implicit class FileOps(val path: Path) extends AnyVal {
+
+    /**
+     * Copies the source to the target, failing if source does not exist or the
+     * target already exists. To replace the existing instead, use
+     * `source.copy(CopyFlags(CopyFlag.ReplaceExisting))(target)`.
+     */
+    def copy(to: Path): IO[Unit] = files.copy(path, to)
+
+    /**
+     * Copies the source to the target, following any directives supplied in the
+     * flags. By default, an error occurs if the target already exists, though
+     * this can be overriden via CopyFlag.ReplaceExisting.
+     */
+    def copy(flags: CopyFlags)(to: Path): IO[Unit] = files.copy(path, to, flags)
+
+    /**
+     * Creates the specified directory. Fails if the parent path does not
+     * already exist.
+     */
+    def createDirectory: IO[Unit] = files.createDirectories(path)
+
+    /**
+     * Creates the specified directory with the specified permissions. Fails if
+     * the parent path does not already exist.
+     */
+    def createDirectory(permissions: Option[Permissions]): IO[Unit] =
+      files.createDirectories(path, permissions)
+
+    /**
+     * Creates the specified directory and any non-existant parent directories.
+     */
+    def createFile: IO[Unit] = files.createFile(path)
+
+    /**
+     * Creates the specified directory and any parent directories, using the
+     * supplied permissions for any directories that get created as a result of
+     * this operation. For example if `/a` exists and
+     * `createDirectories(Path("/a/b/c"), Some(p))` is called, `/a/b` and
+     * `/a/b/c` are created with permissions set to `p` on each (and the
+     * permissions of `/a` remain unmodified).
+     */
+    def createFile(permissions: Option[Permissions]): IO[Unit] =
+      files.createFile(path, permissions)
+
+    /** Creates a hard link with an existing file. */
+    def createLink(existing: Path): IO[Unit] = files.createLink(path, existing)
+
+    /** Creates a symbolic link which points to the supplied target. */
+    def createSymbolicLink(target: Path): IO[Unit] =
+      files.createSymbolicLink(path, target)
+
+    /**
+     * Creates a symbolic link which points to the supplied target with optional
+     * permissions.
+     */
+    def createSymbolicLink(
+        target: Path,
+        permissions: Option[Permissions]
+    ): IO[Unit] =
+      files.createSymbolicLink(path, target, permissions)
+
+    /**
+     * Deletes the file/directory if it exists, returning whether it was
+     * deleted.
+     */
+    def deleteIfExists: IO[Boolean] = files.deleteIfExists(path)
+
+    /** Deletes the file/directory recursively, following symbolic links. */
+    def deleteRecursively(followLinks: Boolean = false): IO[Unit] =
+      files.deleteRecursively(path, followLinks)
```

Comment:
can you create 2 different methods without using default args? They're not great for bin compat and future evolutions. 
```suggestion
    def deleteRecursively: IO[Unit] =
      files.deleteRecursively(path, false)
      
    def deleteRecursively(followLinks: Boolean): IO[Unit] =
      files.deleteRecursively(path, followLinks)
```

Reply:
oh good to know this tip